py3status/modules/diskdata.py

```diff
@@ -66,19 +71,36 @@ class Py3status:
 
     # available configuration parameters
     cache_timeout = 10
-    disk = None
+    disks = []
     format = "{disk}: {used_percent}% ({total})"
     format_rate = "[\?min_length=11 {value:.1f} {unit}]"
     format_space = "[\?min_length=5 {value:.1f}]"
-    sector_size = 512
     si_units = False
     thresholds = {
         "free": [(0, "bad"), (10, "degraded"), (100, "good")],
         "total": [(0, "good"), (1024, "degraded"), (1024 * 1024, "bad")],
         "used_percent": [(0, "good"), (40, "degraded"), (75, "bad")],
     }
+    types = ["disks"]
     unit = "B/s"
 
+    class Meta:
+        deprecated = {
+            "remove": [{"param": "sector_size", "msg": "obsolete parameter"}],
+            "rename": [
+                {"param": "disk", "new": "disks", "msg": "rename parameter to `disks`"}
+            ],
+        }
+
+    def _resolve_disk_name(self, path):
+        disk_path = os.path.realpath(path)
+        return os.path.basename(disk_path)
+
+    def _add_monitored_disk(self, path):
+        basename = self._resolve_disk_name(path)
+        with open("/sys/block/{}/queue/hw_sector_size".format(basename)) as ss:
+            self._disks[basename] = int(ss.read().strip())
```

Comment:
Do you confirm you need `strip()` here? This throws error when I have `sda1` in my `disks`.
* `/sys/block/sda/queue/hw_sector_size:512`
```python
Exception in `test_module` post_config_hook() (FileNotFoundError) module.py line 142.
Traceback
FileNotFoundError: [Errno 2] No such file or directory: '/sys/block/sda1/queue/hw_sector_size'
  File "py3status/py3status/module.py", line 142, in prepare_module
    self.module_class.post_config_hook()
  File "diskdata.py", line 175, in post_config_hook
    self._add_monitored_disk("/dev/" + disk)
  File "diskdata.py", line 101, in _add_monitored_disk
    with open("/sys/block/{}/queue/hw_sector_size".format(basename)) as ss:
```

Reply:
it's because looks for the wrong entry, I'm fixing this

Reply:
fixed

Reply:
Can you work on this some more to deal nonexistent devices (i.e. removable or invalid device)? Otherwise, same exception above.

Reply:
fallbacking to 'all' when no avail devices